CacheWarmer/DoctrineMetadataCacheWarmer.php

```diff
@@ -52,7 +54,7 @@ protected function doWarmUp($cacheDir, ArrayAdapter $arrayAdapter): bool
 
         if (method_exists($metadataFactory, 'setCache')) {
             $metadataFactory->setCache($arrayAdapter);
-        } else {
+        } elseif (class_exists(DoctrineProvider::class) && $metadataFactory instanceof AbstractClassMetadataFactory) {
```

Comment:
Why is that condition necessary? are there 3 cases instead of just 2?

Reply:
I started with level 4 and you get this error there:

ERROR: UndefinedInterfaceMethod - CacheWarmer/DoctrineMetadataCacheWarmer.php:59:31 - Method Doctrine\Persistence\Mapping\ClassMetadataFactory::setCacheDriver does not exist (see https://psalm.dev/181)

But I have removed class_exists at least, it no longer seems necessary